src/main/java/me/qoomon/gitversioning/GitUtil.java

```diff
@@ -49,6 +51,15 @@ public static String revParse(Repository repository, String revstr) {
         return rev.getName();
     }
 
+    public static Date revDate(Repository repository, String revstr) {
+        ObjectId rev = unchecked(() -> repository.resolve(revstr));
+        if (rev == null) {
```

Comment:
I think you don't need that null check, but if its needed please throw an IllegalArgumentException or something similar and ensure to never return null.

Reply:
Hi, thank you for the review.
I think `null` is a possibility. In fact in your method `GitUtil#revParse(...)` you are checking for `null` and returning the `NO_COMMIT` constant if that's the case.
In my case I'm returning a `Date`, so it is harder to find a default value to set when the `ObjectId` is null.

I guess there are 2 options:

- `IllegalArgumentException`: Is this what you prefer? It means that if the `ObjectId` resolves to `null` the build will fail...but the behaviour would be inconsistent with the one implemented in the `GitUtil#revParse(...)`.
- Return an Optional<Date>.

I changed the code to use `Optional` and see if it is OK. Otherwise I can revert it back and throw the IllegalArgumentException.

Reply:
I agree. Could you revert just revert your last commit, I prefer to return 'null' (unknown) in case of an empty repository (should be the only case where no date is available)

Reply:
Thanks for the comments, they make sense. Using the commit date probably made things cleaner because it is expressed in seconds since epoch. At this point I can return 0 if the `ObjectId` is null.
I made all the changes you suggested with some renaming due to the switch from author date to commit timestamp.
If you are happy with the changes, it would be great to have a release with the new functionality